### ISSUE TITLE
Disable text shadow for dark text

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#9504]
## Problem

`minetest.colorize(#000000)` isn't very useful because black text doesn't look nice with a black shadow.

## Solutions

Automatically hide the shadow for text below a certain “darkness” threshold. That is, if the main text is considered too dark, no shadow is rendered. Otherwise, the shadow is rendered.

The exact value could be specified in a `minetest.conf` setting.

## Alternatives

Implement enable/disabling of shadows via formspec style (I'm not sure about this).

## Note
I'm not 100% sure about my own suggestions, to be honest. My solutions might not be ideal, feel free to post better ideas if you have them.

**Comment by sorcerykid** [comment]
Agreed, I've often found shadows to be a nuisance when designing forms with darker text. I would suggest using an #RRGGBBAA color code in the same way as for colorizing text. That would allow the greatest flexibility, rather than just disabling shadows altogether.

One possibility might be a new API function such as `minetest.colorize_bg()` that allows for changing the text background color for a given string. Another option would be to add a third parameter to `minetest.colorize()` that specifies which color component of the text to change, either "fg" for the foreground or "bg" for the background. Those are just ideas off the top of my head.

**Comment by paramat** [comment]
> Automatically hide the shadow for text below a certain “darkness” threshold.[...]
The exact value could be specified in a minetest.conf setting.

I do not think a setting is necessary, a good threshold can be chosen.

**Comment by rubenwardy** [comment]
Settings should be avoided where possible. For formspecs, I would prefer a formspec style for this. This wouldn't work with the chat however, if that is a desired feature

**Comment by v-rob** [comment]
I don't know if a style or a setting is necessary.  I don't see many places where keeping the shadow would be beneficial for dark text.  Since this is really an accessibility issue, settings would be better than style, but I don't see the point of either.

**Comment by appgurueu** [comment]
We could also consider a lighter shadow / outline to make dark text legible against a dark background (black chat messages at night, for example). We already have `minetest.get_background_escape_sequence`, it might be made to work in all cases. Escape sequences for custom shadow colors might be handy as well.
